examples/api-samples/src/browser/view/sample-unclosable-view-contribution.ts

```diff
@@ -33,10 +47,46 @@ export class SampleUnclosableViewContribution extends AbstractViewContribution<S
             }
         });
     }
+
+    protected withWidget<T>(widget: Widget | undefined = this.tryGetWidget(), cb: (sampleView: SampleViewUnclosableView) => T): T | false {
+        if (widget instanceof SampleViewUnclosableView && widget.id === SampleViewUnclosableView.ID) {
+            return cb(widget);
+        }
+        return false;
+    }
+
+    registerCommands(registry: CommandRegistry): void {
+        super.registerCommands(registry);
+        registry.registerCommand(SampleToolBarCommand, {
+            execute: () => {
+                this.updateToolbarItemState();
+                this.messageService.info(`${SampleToolBarCommand.label} is toggled = ${this.toolbarItemState}`);
+            },
+            isEnabled: widget => this.withWidget(widget, () => true),
+            isVisible: widget => this.withWidget(widget, () => true),
+            isToggled: () => this.toolbarItemState
+        });
+    }
+
+    async registerToolbarItems(toolbarRegistry: TabBarToolbarRegistry): Promise<void> {
+        toolbarRegistry.registerItem({
+            id: SampleToolBarCommand.id,
+            command: SampleToolBarCommand.id,
+            tooltip: 'Click to Toggle Toolbar Item',
+            priority: 0,
+            onDidChange: this.onSampleToolbarStateChangedEmitter.event
+        });
+    }
+
+    protected updateToolbarItemState(): void {
+        this.toolbarItemState = !this.toolbarItemState;
+        this.onSampleToolbarStateChangedEmitter.fire(undefined);
+    }
```

Comment:
We can either drop the method altogether (and place it in the `execute`), or we can drop the `fire()`.